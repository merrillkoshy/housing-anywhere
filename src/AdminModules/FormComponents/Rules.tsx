import React from "react";
import { Form, Col, Row } from "react-bootstrap";
import Switch from "react-switch";

const Rules = ({
	pets,
	setPets,
	cleaningCommonRoomsIncluded,
	setCleaningCommonRoomsIncluded,
	cleaningPrivateRoomIncluded,
	setCleaningPrivateRoomIncluded,
	playMusic,
	setPlayMusic,
	smoking,
	setSmoking,
	couple,
	setCouple,
}: {
	pets: string;
	setPets: React.Dispatch<React.SetStateAction<string>>;
	cleaningCommonRoomsIncluded: boolean;
	setCleaningCommonRoomsIncluded: React.Dispatch<React.SetStateAction<boolean>>;
	cleaningPrivateRoomIncluded: boolean;
	setCleaningPrivateRoomIncluded: React.Dispatch<React.SetStateAction<boolean>>;
	playMusic: string;
	setPlayMusic: React.Dispatch<React.SetStateAction<string>>;
	smoking: string;
	setSmoking: React.Dispatch<React.SetStateAction<string>>;
	couple: boolean;
	setCouple: React.Dispatch<React.SetStateAction<boolean>>;
}) => {
	return (
		<>
			<Form.Group className="mb-3">
				<Form.Label>Pets</Form.Label>
				<Form.Control
					type="text"
					id="pets"
					value={pets}
					onChange={(e) => {
						e.preventDefault();
						setPets(e.currentTarget.value);
					}}
					placeholder="Allowed?"
				/>
			</Form.Group>

			<Form.Group as={Row} className="mb-3">
				<Form.Label as="legend" column>
					Cleaning Common Rooms Included ?
				</Form.Label>
				<Col>
					<Switch
						id="cleaningCommonRoomsIncluded"
						onChange={() => {
							setCleaningCommonRoomsIncluded(!cleaningCommonRoomsIncluded);
						}}
						checked={cleaningCommonRoomsIncluded}
					/>
				</Col>
			</Form.Group>
			<Form.Group as={Row} className="mb-3">
				<Form.Label as="legend" column>
					Cleaning Private Rooms Included ?
				</Form.Label>
				<Col>
					<Switch
						id="cleaningPrivateRoomIncluded"
						onChange={() => {
							setCleaningPrivateRoomIncluded(!cleaningPrivateRoomIncluded);
						}}
						checked={cleaningPrivateRoomIncluded}
					/>
				</Col>
			</Form.Group>
			<Form.Group className="mb-3">
				<Form.Label>Smoking Allowed?</Form.Label>
				<Form.Control
					type="text"
					id="smoking"
					value={smoking}
					onChange={(e) => {
						e.preventDefault();
						setSmoking(e.currentTarget.value);
					}}
					placeholder="Allowed?"
				/>
			</Form.Group>
			<Form.Group className="mb-3">
				<Form.Label>Can Play Musical instruments?</Form.Label>
				<Form.Control
					type="text"
					id="playMusic"
					value={playMusic}
					onChange={(e) => {
						e.preventDefault();
						setPlayMusic(e.currentTarget.value);
					}}
					placeholder="Allowed?"
				/>
			</Form.Group>
			<Form.Group as={Row} className="mb-3">
				<Form.Label as="legend" column>
					Couples Allowed?
				</Form.Label>
				<Col>
					<Switch
						id="couple"
						onChange={() => {
							setCouple(!couple);
						}}
						checked={couple}
					/>
				</Col>
			</Form.Group>
		</>
	);
};
export default Rules;
