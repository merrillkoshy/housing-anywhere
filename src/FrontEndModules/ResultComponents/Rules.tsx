import React from "react";
import { Rules } from "../../components/dataInterfaces";

const RulesComponent = ({ rules }: { rules: Rules }) => {
	return (
		<details>
			<summary>
				<strong>Rules</strong>
			</summary>
			<ul>
				<li>Pets:{rules?.pets === "yes" ? "✅" : "❌"}</li>
				<li>
					Cleaning Common Rooms Included:
					{rules?.cleaningCommonRoomsIncluded ? "✅" : "❌"}
				</li>
				<li>
					Cleaning Private Rooms Included:
					{rules?.cleaningPrivateRoomIncluded ? "✅" : "❌"}
				</li>
				<li>Play Music:{rules?.playMusic === "yes" ? "✅" : "❌"}</li>
				<li>Smoking:{rules?.smoking === "yes" ? "✅" : "❌"}</li>
				<li>Couple:{rules?.couple ? "✅" : "❌"}</li>
			</ul>
		</details>
	);
};
export default RulesComponent;
