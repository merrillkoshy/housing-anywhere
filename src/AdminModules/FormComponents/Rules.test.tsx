import React, { SetStateAction } from "react";
import { shallow } from "enzyme";
import Rules from "./Rules";

describe("Address module", () => {
	const RulesModule = shallow(
		<Rules
			pets={""}
			setPets={function(value: SetStateAction<string>): void {
				throw new Error("Function not implemented.");
			}}
			cleaningCommonRoomsIncluded={false}
			setCleaningCommonRoomsIncluded={function(
				value: SetStateAction<boolean>
			): void {
				throw new Error("Function not implemented.");
			}}
			cleaningPrivateRoomIncluded={false}
			setCleaningPrivateRoomIncluded={function(
				value: SetStateAction<boolean>
			): void {
				throw new Error("Function not implemented.");
			}}
			playMusic={""}
			setPlayMusic={function(value: SetStateAction<string>): void {
				throw new Error("Function not implemented.");
			}}
			smoking={""}
			setSmoking={function(value: SetStateAction<string>): void {
				throw new Error("Function not implemented.");
			}}
			couple={false}
			setCouple={function(value: SetStateAction<boolean>): void {
				throw new Error("Function not implemented.");
			}}
		/>
	);
	it("renders", () => {
		expect(RulesModule).toMatchSnapshot();
	});
	it("finds a pets input which is a string", () => {
		expect(RulesModule.find("#pets").exists()).toBe(true);
	});
	it("finds a Cleaning CommonRooms Included input which is a bool", () => {
		expect(RulesModule.find("#cleaningCommonRoomsIncluded").exists()).toBe(
			true
		);
	});
	it("finds a Cleaning Private RoomIncluded input which is a bool", () => {
		expect(RulesModule.find("#cleaningPrivateRoomIncluded").exists()).toBe(
			true
		);
	});
	it("finds a Play Music input which is a string", () => {
		expect(RulesModule.find("#playMusic").exists()).toBe(true);
	});
	it("finds a smoking input which is a string", () => {
		expect(RulesModule.find("#smoking").exists()).toBe(true);
	});
	it("finds a couple input which is a bool", () => {
		expect(RulesModule.find("#couple").exists()).toBe(true);
	});
});
