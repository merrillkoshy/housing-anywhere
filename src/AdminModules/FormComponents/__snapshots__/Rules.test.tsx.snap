// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Address module renders 1`] = `
<Fragment>
  <FormGroup
    className="mb-3"
  >
    <FormLabel
      column={false}
      srOnly={false}
    >
      Pets
    </FormLabel>
    <FormControl
      id="pets"
      onChange={[Function]}
      placeholder="Allowed?"
      type="text"
      value=""
    />
  </FormGroup>
  <FormGroup
    as={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": Object {
          "noGutters": false,
        },
        "render": [Function],
      }
    }
    className="mb-3"
  >
    <FormLabel
      as="legend"
      column={true}
      srOnly={false}
    >
      Cleaning Common Rooms Included ?
    </FormLabel>
    <Col>
      <ReactSwitch
        activeBoxShadow="0 0 2px 3px #3bf"
        boxShadow={null}
        checked={false}
        checkedIcon={
          <svg
            height="100%"
            style={
              Object {
                "position": "absolute",
                "top": 0,
              }
            }
            viewBox="-2 -5 17 21"
            width="100%"
          >
            <path
              d="M11.264 0L5.26 6.004 2.103 2.847 0 4.95l5.26 5.26 8.108-8.107L11.264 0"
              fill="#fff"
              fillRule="evenodd"
            />
          </svg>
        }
        disabled={false}
        height={28}
        id="cleaningCommonRoomsIncluded"
        offColor="#888"
        offHandleColor="#fff"
        onChange={[Function]}
        onColor="#080"
        onHandleColor="#fff"
        uncheckedIcon={
          <svg
            height="100%"
            style={
              Object {
                "position": "absolute",
                "top": 0,
              }
            }
            viewBox="-2 -5 14 20"
            width="100%"
          >
            <path
              d="M9.9 2.12L7.78 0 4.95 2.828 2.12 0 0 2.12l2.83 2.83L0 7.776 2.123 9.9 4.95 7.07 7.78 9.9 9.9 7.776 7.072 4.95 9.9 2.12"
              fill="#fff"
              fillRule="evenodd"
            />
          </svg>
        }
        width={56}
      />
    </Col>
  </FormGroup>
  <FormGroup
    as={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": Object {
          "noGutters": false,
        },
        "render": [Function],
      }
    }
    className="mb-3"
  >
    <FormLabel
      as="legend"
      column={true}
      srOnly={false}
    >
      Cleaning Private Rooms Included ?
    </FormLabel>
    <Col>
      <ReactSwitch
        activeBoxShadow="0 0 2px 3px #3bf"
        boxShadow={null}
        checked={false}
        checkedIcon={
          <svg
            height="100%"
            style={
              Object {
                "position": "absolute",
                "top": 0,
              }
            }
            viewBox="-2 -5 17 21"
            width="100%"
          >
            <path
              d="M11.264 0L5.26 6.004 2.103 2.847 0 4.95l5.26 5.26 8.108-8.107L11.264 0"
              fill="#fff"
              fillRule="evenodd"
            />
          </svg>
        }
        disabled={false}
        height={28}
        id="cleaningPrivateRoomIncluded"
        offColor="#888"
        offHandleColor="#fff"
        onChange={[Function]}
        onColor="#080"
        onHandleColor="#fff"
        uncheckedIcon={
          <svg
            height="100%"
            style={
              Object {
                "position": "absolute",
                "top": 0,
              }
            }
            viewBox="-2 -5 14 20"
            width="100%"
          >
            <path
              d="M9.9 2.12L7.78 0 4.95 2.828 2.12 0 0 2.12l2.83 2.83L0 7.776 2.123 9.9 4.95 7.07 7.78 9.9 9.9 7.776 7.072 4.95 9.9 2.12"
              fill="#fff"
              fillRule="evenodd"
            />
          </svg>
        }
        width={56}
      />
    </Col>
  </FormGroup>
  <FormGroup
    className="mb-3"
  >
    <FormLabel
      column={false}
      srOnly={false}
    >
      Smoking Allowed?
    </FormLabel>
    <FormControl
      id="smoking"
      onChange={[Function]}
      placeholder="Allowed?"
      type="text"
      value=""
    />
  </FormGroup>
  <FormGroup
    className="mb-3"
  >
    <FormLabel
      column={false}
      srOnly={false}
    >
      Can Play Musical instruments?
    </FormLabel>
    <FormControl
      id="playMusic"
      onChange={[Function]}
      placeholder="Allowed?"
      type="text"
      value=""
    />
  </FormGroup>
  <FormGroup
    as={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": Object {
          "noGutters": false,
        },
        "render": [Function],
      }
    }
    className="mb-3"
  >
    <FormLabel
      as="legend"
      column={true}
      srOnly={false}
    >
      Couples Allowed?
    </FormLabel>
    <FormControl
      id="couple"
      onChange={[Function]}
      placeholder="Allowed?"
      type="text"
      value={false}
    />
  </FormGroup>
</Fragment>
`;
